arrow/src/main/java/org/apache/iceberg/arrow/vectorized/GenericArrowVectorAccessorFactory.java

```diff
@@ -244,6 +248,18 @@ public final boolean getBoolean(int rowId) {
     }
   }
 
+  private static class NullAccessor<
+          DecimalT, Utf8StringT, ArrayT, ChildVectorT extends AutoCloseable>
+      extends ArrowVectorAccessor<DecimalT, Utf8StringT, ArrayT, ChildVectorT> {
+
+    private final NullVector vector;
```

Comment:
I don't think we need to track this here separately, since the super class already has a `vector`